ext/oss/sink.go

```diff
@@ -193,14 +200,44 @@ func (o *OSSSink) process() error {
 		if recordCounter%logCheckPoint == 0 {
 			o.Logger().Info(fmt.Sprintf("written %d records to tmp file: %s", o.fileRecordCounters[tmpPath], tmpPath))
 		}
+
+		// EXPERIMENTAL: flush and upload to OSS if maximum number of records in tmp file is reached.
+		// a method of partial upload is implemented to avoid having large number of tmp files which leads to high disk usage.
+		// as long as the streamed records from source are guaranteed to be uniform (same header & value ordering), this should work.
+		if o.fileRecordCounters[tmpPath] >= MAX_TEMP_FILE_RECORD_NUMBER {
+			o.Logger().Info(fmt.Sprintf("maximum number of temp file records %d reached. flushing %s to OSS", MAX_TEMP_FILE_RECORD_NUMBER, tmpPath))
+			if err := wh.Flush(); err != nil {
+				o.Logger().Error(fmt.Sprintf("failed to flush tmp file: %s", tmpPath))
+				return errors.WithStack(err)
+			}
+			o.Logger().Info(fmt.Sprintf("flushed tmp file: %s", tmpPath))
+
+			if err := o.flush(destinationURI, o.ossHandlers[destinationURI]); err != nil {
+				o.Logger().Error(fmt.Sprintf("failed to upload to OSS: %s", destinationURI))
+				return errors.WithStack(err)
+			}
+
+			// reset counters and handlers for the current batch
+			// except for oss handler which is reused to upload the next part of the file
+			o.fileRecordCounters[tmpPath] = 0
```

Comment:
can we use `partiallyUploadedFiles` counter instead to track the records number if it reach max? and reset that counter? `o.fileRecordCounters` is used to track overall records count for particular file

Reply:
let me check